docs/developer/best_practices/commit.rst

```diff
@@ -0,0 +1,372 @@
+a change might seem trivial when you write it but will seem totally awkward
+the day you will have to understand why you made it. If you think your patch is
+trivial and somebody tells you he does not understand your patch, then your
+patch is not trivial and it requires a detailed description.
+
+The footer contains references for issue management (``Refs``, ``Closes``, …)
+or other relevant annotations (cherry-pick source, …).
+Optional if your commit is not related to any issue (should be pretty rare).
+
+Subject
+:::::::
+
+A good commit message should start with a short summary of the change: the
+subject line.
+
+This summary should be written using the imperative mood and carry as much
+information as possible while staying short, ideally under 50 characters (this
+is a goal, the hard limit is 72).
+
+Subject topic and description shouldn't start with a capital.
+
+It is composed of:
+
+- a topic, usually the name of the affected component (``ui``, ``build``,
+  ``docs``, …)
+
+- a slash and then the name of the sub-component (optional)
+
+- a colon
+
+- the description of the change
+
+Examples:
+
+- ``ci: use proxy-cache to reduce flakiness``
+
+- ``build/package: factorize task_dep in DEBPackage``
+
+- ``ui/volume: add banner when failed to create volume``
+
+If several components are affected:
+
+- split your commit (preferred)
+
+- pick only the most affected one
+
+- entirely omit the component (happen for truly global change, like renaming
+  ``licence`` to ``license`` over the whole codebase)
+
+As for "what is the topic?", the following heuristic works quite well for
+MetalK8s: take the name of the top-level directory (``ui``, ``salt``, ``docs``,
+…) except for ``eve`` (use ``ci`` instead). ``buildchain`` could also be
+shortened to ``build``.
+
+Having the topic in the summary line allows for faster peering over ``git log``
+output (you can know what the commit is about just by reading a few characters,
+not need to check the entire commit message or the associated diff).
+It also helps the review process: if you have a big pull request affecting
+front-end and back-end, front-end people can only review commits starting with
+``ui`` (not need to read over the whole diff, or to open each commit one by one
+in Github to see which ones are interesting).
+
+Body
+::::
+
+The body should answer the following questions:
+
+- Why did you make this change? (is this for a new feature, a bugfix — then,
+  why was it buggy? —, some cleanup, some optimization, …).
+  It is really important to describe the intent/motivation behind the changes.
+
+- What change did you make? Document what the original problem was and how it
+  is being fixed (can be omitted for short obvious patches).
+
+- Why did you make the change in that way and not in another (mention alternate
+  solutions considered but discarded, if any)?
+
+When writing your message you must consider that your reader does not know
```

Comment:
I'd somewhat reduce this statement: it's almost impossible to write _"noob-friendly"_ commits while carrying out meaningful information for e.g. knowledgeable reviewers, without writing full pages for each change.

Maybe we could go for something closer to "Try to write your message in an explicit manner, adding references when concepts would need explanation. If the reader comes from outside of the core developers, she needs to be able to navigate through to understand the change".

Reply:
It's not about noob-friendliness, as in "explain acronyms such as IPIP" or "what is a Pod?".
It's more about anyone should be able to review the code, even if it's not in their comfort zone.

That kinda goes hand in hand with the bullet points above: give context and explain why such as an outsider understand the commit.